pkg/kubelet/kubelet.go

```diff
@@ -2105,9 +2105,15 @@ func (kl *Kubelet) SyncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 		semconv.K8SNamespaceNameKey.String(pod.Namespace),
 	))
 	logger := klog.FromContext(ctx)
-	defer otelSpan.End()
 	klog.V(4).InfoS("SyncTerminatingPod enter", "pod", klog.KObj(pod), "podUID", pod.UID)
-	defer klog.V(4).InfoS("SyncTerminatingPod exit", "pod", klog.KObj(pod), "podUID", pod.UID)
+	defer func() {
+		if err != nil {
+			otelSpan.RecordError(err)
+			otelSpan.SetStatus(codes.Error, err.Error())
+		}
+		klog.V(4).InfoS("SyncTerminatingPod exit", "pod", klog.KObj(pod), "podUID", pod.UID)
```

Comment:
whty logger is not used here and a few methods below?

Reply:
I want to see the `kl.tracer.Start(` call directly followed by  `defer func() {`. Logger initialization and the log message can go before it. Also you can use logger here as an optional improvement. But I would understand if this will be a separate PR later.